packages/video_player/android/src/main/java/io/flutter/plugins/videoplayer/VideoPlayerPlugin.java

```diff
@@ -107,9 +108,35 @@ private static boolean isFileOrAsset(Uri uri) {
       return scheme.equals("file") || scheme.equals("asset");
     }
 
+    private static final FORMAT_SS = "ss";
+    private static final FORMAT_DASH = "dash";
+    private static final FORMAT_HLS = "hls";
+    private static final FORMAT_OTHER = "other";
+
     private MediaSource buildMediaSource(
-        Uri uri, DataSource.Factory mediaDataSourceFactory, Context context) {
-      int type = Util.inferContentType(uri.getLastPathSegment());
+        Uri uri, DataSource.Factory mediaDataSourceFactory, String formatHint, Context context) {
+      int type;
```

Comment:
nits maybe:
```java
      if (formatHint != null) {
        switch (formatHint) {
          case FORMAT_SS:
            type = C.TYPE_SS;
            break;
          case FORMAT_DASH:
            type = C.TYPE_DASH;
            break;
          case FORMAT_HLS:
            type = C.TYPE_HLS;
            break;
          case FORMAT_OTHER:
            type = C.TYPE_OTHER;
            break;
          default:
            type = -1;
            break;
        }
      } else {
        int type = Util.inferContentType(uri.getLastPathSegment());
        switch (type) {...
     }
```

Reply:
GitHub swallowed my comment:

You've moved the swtich() statement into the else. This would mean formathint wouldn't return the MediaSource and instead return null.